analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java

```diff
@@ -644,4 +711,37 @@ private void writeRepositorySettingsFile(Map<ArtifactRepository, Path> repositor
         return allDependencies;
     }
 
+    public void writeReport(File rootDir, String report) throws ManipulationException {
+        final File reportFile = new File(".", "report-pme-"
+                + RandomStringUtils.randomAlphanumeric(10) + ".txt");
+
+        try {
+            FileUtils.writeStringToFile(reportFile, report, StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new ManipulationException("Unable to write " + reportFile + " in project root", e);
+        }
+    }
+
+    public void writeJsonModel(File rootDir, PME jsonModel) throws ManipulationException {
+        final File manipulationFile = new File(".", MANIPULATION_PME_BASENAME + "-"
+                + RandomStringUtils.randomAlphanumeric(10) + ".json");
+
+        try {
+            FileUtils.writeStringToFile(manipulationFile, JSONUtils.jsonToString(jsonModel), StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new ManipulationException("Unable to write " + manipulationFile + " in project root", e);
+        }
+    }
+
+    /**
+     * Wraps appending to the string builder using SLF4J style substitutions.
+     *
+     * @param builder the string builder
+     * @param message the message (possibly with parameters)
+     * @param args optional parameters.
+     */
+    private static void append(StringBuilder builder, String message, Object... args) {
+        builder.append(MessageFormatter.arrayFormat(message, args).getMessage());
+        builder.append('\n');
+    }
```

Comment:
Nice, although I do wonder if it would be better in common?

Reply:
Yes, I just copied this one from PME (more or less).

There is an issue with what separator to use, since normally if we print this to the console we would use `System.lineSeparator()`, but if we want to save it to a text file, it might be better to normalize it to `\n` so that it produces the exact content across platforms (of course, we won't hit this in practice, so it's not that crucial).

If we were able to print it on the fly, we could just call `log`, but I guess PME is collecting it as a string just to be able to have all the content available for a file later?